pkg/tangy/rpm.go

```diff
@@ -353,6 +368,88 @@ func (t *tangyImpl) RpmRepositoryVersionErrataList(ctx context.Context, hrefs []
 	return errata, countTotal, nil
 }
 
+// RpmRepositoryVersionModuleStreamsList List Modules streams within a repository version, with pagination, search and an optional name filter
+func (t *tangyImpl) RpmRepositoryVersionModuleStreamsList(ctx context.Context, hrefs []string, filterOpts ModuleStreamListFilters, sortBy string) ([]ModuleStreams, int, error) {
+	if len(hrefs) == 0 {
+		return []ModuleStreams{}, 0, nil
+	}
+
+	conn, err := t.pool.Acquire(ctx)
+	if err != nil {
+		return nil, 0, err
+	}
+	defer conn.Release()
+
+	repoVerMap, err := parseRepositoryVersionHrefsMap(hrefs)
+	if err != nil {
+		return nil, 0, fmt.Errorf("error parsing repository version hrefs: %w", err)
+	}
+
+	orderBy := ""
+	if SortBy == "" {
+		orderBy = "rp.name, rp.stream"
+	} else {
+		switch true {
+		case strings.Contains(SortBy, "name"):
+			orderBy = "rp.name"
+		case strings.Contains(SortBy, "stream"):
+			orderBy = "rp.stream"
+		default:
+			orderBy = ""
+		}
+
+		if strings.Contains(SortBy, "desc") {
+			orderBy += " DESC"
+		} else {
+			orderBy += " ASC"
+		}
+	}
+
+	args := pgx.NamedArgs{
+		"nameFilter": "%" + filterOpts.Search + "%",
+		"rpm_names":  filterOpts.RpmNames,
+	}
+	query := `SELECT rp.name, rp.stream, max(rp.version) as version, rp.profiles, rp.context, rp.arch, rp.description FROM rpm_modulemd rp 
+	INNER JOIN rpm_modulemd_packages rmp on rmp.modulemd_id = rp.content_ptr_id
+	INNER JOIN rpm_package pack on pack.content_ptr_id = rmp.package_id `
+
+	countQueryOpen := `SELECT COUNT(*) FROM (Select distinct name from (` + query
+	innerUnion := contentIdsInVersions(repoVerMap, &args)
+
+	rpmNameFilter := ""
+
+	if len(filterOpts.RpmNames) > 0 {
+		rpmNameFilter = " AND pack.name = ANY(@rpm_names)"
+	}
+
+	filter := rpmNameFilter + " AND rp.name ILIKE CONCAT( '%', @nameFilter::text, '%') GROUP BY rp.name, rp.stream, rp.context,rp.profiles, rp.arch, rp.description"
```

Comment:
I'm not sure that description, context, profiles will necessarily alwasy be the same, so group by may not work here.  

You might try something like:

```
 SELECT DISTINCT ON (name, stream) 
    uuid, 
    name, 
    stream, 
    version, 
    context
FROM rpm_modulemd
ORDER BY name, stream, version DESC;
```

with all the joins and such

Reply:
:)